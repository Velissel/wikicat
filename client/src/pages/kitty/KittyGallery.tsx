import React from 'react';
import { Col, Image, Row } from 'react-bootstrap';
// @ts-ignore
import styles from './KittyGallery.module.scss';
import { isEmpty } from 'lodash';

interface KittyGalleryProps {
  images: string[];
}
export default function KittyGallery(props: KittyGalleryProps) {
  const { images } = props;

  if (isEmpty(images)) {
    return <h4>No other photos</h4>;
  }

  return (
    <Row>
      {images.map((image, i) => {
        return (
          <Col xs={6} sm={4} md={4} lg={3} key={image}>
            <Image
              src={image}
              fluid
              rounded
              className={`mb-4 ${styles.image}`}
              data-testid={`img-${i}`}
            ></Image>
          </Col>
        );
      })}
    </Row>
  );
}
