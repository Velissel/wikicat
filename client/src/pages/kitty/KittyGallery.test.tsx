import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import KittyGallery from './KittyGallery';

describe('KittyGallery component', () => {
  it('should render text when no photo is supplied', () => {
    const images = [];
    const { container } = render(
      <BrowserRouter>
        <KittyGallery images={images} />
      </BrowserRouter>,
    );

    expect(screen.getByText('No other photos')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render images when correct data is supplied', () => {
    const images = ['url_1', 'url_2'];
    const { container } = render(
      <BrowserRouter>
        <KittyGallery images={images} />
      </BrowserRouter>,
    );

    images.forEach((image, i) => {
      expect(screen.getByTestId(`img-${i}`)).toBeInTheDocument();
    });
    expect(container).toMatchSnapshot();
  });
});
