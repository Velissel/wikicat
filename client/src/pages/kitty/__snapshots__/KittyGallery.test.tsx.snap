// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KittyGallery component should render images when correct data is supplied 1`] = `
<div>
  <div
    class="row"
    data-testid="kitty-photos"
  >
    <div
      class="col-lg-3 col-md-4 col-sm-4 col-6"
    >
      <img
        class="mb-4 image img-fluid rounded"
        data-testid="img-0"
        src="url_1"
      />
    </div>
    <div
      class="col-lg-3 col-md-4 col-sm-4 col-6"
    >
      <img
        class="mb-4 image img-fluid rounded"
        data-testid="img-1"
        src="url_2"
      />
    </div>
  </div>
</div>
`;

exports[`KittyGallery component should render text when no photo is supplied 1`] = `
<div>
  <h4>
    No other photos
  </h4>
</div>
`;
